### PR TITLE
test: 40% coverage of team reports page

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at f6053d4</samp>

This pull request adds and updates some files related to the team reports page component, which displays a list of reports that need approval or review by the current user. It enhances the component by adding some missing features, such as sorting, filtering, and searching, and by removing some redundant and unused code. It also adds some mock data, models, and test cases for the component, using the TestBed class and some predefined query parameters.

### <samp>🤖 Generated by Copilot at f6053d4</samp>

> _Team reports page tests_
> _Mock data and models added_
> _Winter of refactor_

### <samp>🤖 Generated by Copilot at f6053d4</samp>

*  Add and modify imports for ionic/angular and core/models types in the team reports page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596L7-R7), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596R26))
*  Change the type of the loadData$ observable in the team reports page component to GetTasksQueryParamsWithFilters ([link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596L57-R58))
*  Simplify the logic for setting the filters and loading the data in the team reports page component by removing unnecessary conditionals and parameters ([link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596L228-R225), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596L214-R211))
*  Add type annotations to the event parameters of the loadData and doRefresh methods in the team reports page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596L265-R242), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596L275-R252), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596L281-R258))
*  Remove an unnecessary optional chaining operator from the event.target.complete call in the doRefresh method in the team reports page component ([link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-3ccc9338e2efb6f5e87dcb962c41933667b5e6a3477accd2534ccb98af447596L281-R258))
*  Add mock data and models for the report and tasks API parameters, with different sorting, filtering, and searching options ([link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-ea3bfa076051ba9c435df7ef1da7334cab954ea91a51c5a33839dae47dc5e3e8R20-R37), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-6fe7c64e8d36575aaa8d32fd7f3b48031688077375d32f158ec8639db22de0aeR1-R16), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-e24f155bcefe0b088ad43a08dd05c6d7290b25dc2479ed65a05a52b8cd4fbe26R1-R7), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-8a648a45db178e1d5321c3bc13b2647712ffe4d072b02143f43e78e4ac8f8c3aR1-R9), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-313e8684d3e0948928af56754bcb29d24a79c605d650b0dc7759d1b5039d0b9aR1-R7))
*  Add a test suite for the team reports page component, using the mock data and models, and a setup function to create and configure a testing module ([link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-36f71f770bc053284d3f41359139a1acea955135e4d922a1b75a2716cbb9f389R1-R378), [link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-35c5d9087f10bb379b3c726e6e332f85a6d6f39dc932ad40ee50015db7164e5aR1-R75))
*  Delete the unused team reports page spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2157/files?diff=unified&w=0#diff-c2032858bc6c823cc5cf86f320c3121fc00192b51e1b292f74c470bd81a52b02))

## Clickup
https://app.clickup.com/t/85zt7qcf7

## Code Coverage
Statements 125/317
Branches 54/172
Functions 25/68
Lines 122/310

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/team-reports/team-reports.page.ts [hunk 1]
```diff
@@ -225,26 +218,11 @@ export class TeamReportsPage implements OnInit {
       });
     });
 
-    if (this.activatedRoute.snapshot.queryParams.filters) {
-      this.filters = Object.assign({}, this.filters, JSON.parse(this.activatedRoute.snapshot.queryParams.filters));
-      this.currentPageNumber = 1;
-      const params = this.addNewFiltersToParams();
-      this.loadData$.next(params);
-      this.filterPills = this.generateFilterPills(this.filters);
-    } else if (this.activatedRoute.snapshot.params.state) {
```
Comment (Julias0) [inline src/app/fyle/team-reports/team-reports.page.ts:234]: this condition dosnt seem to be handled in the block below - is that correct?

File: src/app/fyle/team-reports/team-reports-1.page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+        component.ionViewWillEnter();
+        inputElement.value = 'example';
+        inputElement.dispatchEvent(new Event('keyup'));
+        tick(1000);
+        expect(component.currentPageNumber).toBe(1);
+      }));
+
+      it('should call apiV2Service.extendQueryParamsForTextSearch', () => {
+        component.ionViewWillEnter();
+        expect(apiV2Service.extendQueryParamsForTextSearch).toHaveBeenCalledTimes(4);
```
Comment (Julias0) [inline src/app/fyle/team-reports/team-reports-1.page.spec.ts:201]: why should this method be called 4 times during entry?

**Review by Julias0 — CHANGES_REQUESTED** [review]
few comments

File: src/app/fyle/team-reports/team-reports-1.page.spec.ts (hunk 2 above)
Comment (suyashpatil78) [reply src/app/fyle/team-reports/team-reports-1.page.spec.ts:201]: Reason for 4 calls of extendQueryParamsForTextSearch.
loadData$ is behaviorSubject. We are using pipe on it for:
`teamReports$`
`count$`
When we subscribe to teamReports and count, method gets called twice when loadData$ is set initially and they get called again when we update loadData$.next() therefore making 4 calls.

**Review by Dimple16 — COMMENTED** [review]
Dhar has reviewed it, will skip

**Review by Julias0 — APPROVED** [review]

**Pull request merged**